third_party/bazel/common.bzl

```diff
@@ -0,0 +1,338 @@
+"""Functions common across configure rules."""
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+BAZEL_SH = "BAZEL_SH"
+PYTHON_BIN_PATH = "PYTHON_BIN_PATH"
+PYTHON_LIB_PATH = "PYTHON_LIB_PATH"
+TF_PYTHON_CONFIG_REPO = "TF_PYTHON_CONFIG_REPO"
+
+def auto_config_fail(msg):
+    """Output failure message when auto configuration fails."""
+    red = "\033[0;31m"
+    no_color = "\033[0m"
+    fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
+
+def which(repository_ctx, program_name, allow_failure = False):
+    """Returns the full path to a program on the execution platform.
+
+    Args:
+      repository_ctx: the repository_ctx
+      program_name: name of the program on the PATH
+
+    Returns:
+      The full path to a program on the execution platform.
+    """
+    if is_windows(repository_ctx):
+        if not program_name.endswith(".exe"):
+            program_name = program_name + ".exe"
+        out = execute(
+            repository_ctx,
+            ["C:\\Windows\\System32\\where.exe", program_name],
+            allow_failure = allow_failure,
+        ).stdout
+        if out != None:
+            out = out.replace("\\", "\\\\").rstrip()
+        return out
+
+    out = execute(
+        repository_ctx,
+        ["which", program_name],
+        allow_failure = allow_failure,
+    ).stdout
+    if out != None:
+        out = out.replace("\\", "\\\\").rstrip()
+    return out
+
+def get_python_bin(repository_ctx):
```

Comment:
Nit: why not use `sys.executable ? Such as,

```
>>> import sys
>>> print(sys.executable)
/opt/conda/bin/python3
```

Reply:
This is following what tf has been doing to find python executable, getting **PYTHON_BIN_PATH** setting from bazelrc's action env. When the python we want to use is not default system python, such as a python executable from a virtual env，getting py bin using **sys.executable** would be wrong.